src/Bridges/SymfonyConsole/CreateCommand.php

```diff
@@ -89,7 +141,92 @@ private function hasNumericSubdirectory($dir, & $found)
 				return TRUE;
 			}
 		}
+
 		return FALSE;
 	}
 
+
+	/**
+	 * @return string
+	 */
+	protected function getTypeArgDescription()
+	{
+		$options = [];
+		$groups = $this->config->getGroups();
+		usort($groups, function (Group $a, Group $b) {
+			return strcmp($a->name, $b->name);
+		});
+
+		foreach ($groups as $i => $group) {
+			for ($j = 1; $j < strlen($group->name); $j++) {
+				if (!isset($groups[$i + 1]) || strncmp($group->name, $groups[$i + 1]->name, $j) !== 0) {
+					$options[] = substr($group->name, 0, $j) . '(' . substr($group->name, $j) . ')';
+					break;
+				}
+			}
+		}
+
+		return implode(' or ', array_filter([
+			implode(', ', array_slice($options, 0, -1)),
+			array_slice($options, -1)[0],
+		]));
+	}
+
+
+	/**
+	 * @param  InputInterface $input
+	 * @return string
+	 */
+	protected function getFileContentSource(InputInterface $input)
+	{
+		if ($input->hasOption('diff')) {
+			return self::CONTENT_SOURCE_DIFF;
+
+		} elseif ($input->hasOption('stdin')) {
+			return self::CONTENT_SOURCE_STDIN;
+
+		} elseif ($input->hasOption('empty')) {
+			return self::CONTENT_SOURCE_EMPTY;
+
+		} else {
+			return $this->defaultContentSource;
+		}
+	}
+
+
+	/**
+	 * @param  Group  $group
+	 * @param  string $source
+	 * @return string
+	 */
+	protected function getFileContent(Group $group, $source)
+	{
+		if ($source === self::CONTENT_SOURCE_DIFF && $group->generator !== NULL) {
+			return $group->generator->generateContent();
+
+		} elseif ($source === self::CONTENT_SOURCE_STDIN) {
+			return stream_get_contents(STDIN);
+
+		} else {
+			return '';
```

Comment:
To jde snadno napsat jako vlastní `IDiffGenerator`. Čímž se tak trochu dostáváme k tomu, jestli by to volání `stream_get_contents(STDIN)` nemělo být taky uvnitř nějaké built-in `IDiffGenerator` implementace. A možná pak přejmenovat `IDiffGenerator` jenom na `IGenerator`?